### ISSUE TITLE
feature: extend Response API

### ISSUE THREAD

**Issue by Lancetnik** [airtai/faststream#1587]
We should specify concrete broker Response classes (like [KafkaResponse](https://github.com/airtai/faststream/blob/main/faststream/kafka/response.py#L4)) to make them supporting all broker features ([Publisher.publish](https://github.com/airtai/faststream/blob/main/faststream/kafka/publisher/usecase.py#L100) options)

Another words, **KafkaResponse** should has the following signature:

```python
class KafkaResponse(Response):
    def __init__(
        self,
        body: "SendableMessage",
        *,
        headers: Optional["AnyDict"] = None,
        correlation_id: Optional[str] = None,
        # Kafka specific
        timestamp_ms: Optional[int] = None,
        key: Optional[bytes] = None,
        ...
    ) -> None:
        ...
```

Another brokers Response classes should be extended the same way

**Comment by Flosckow** [comment]
Do I understand correctly that it is necessary to change all "SpecificBroker".Response classes, transfer variables that are in the "SpecificBroker".publisher.use case to the init method ("SpecificBroker".Response)?
And why is the **body: "SendableMessage"** parameter specified in **Response**, and in the **publish** methods we have **message: "SendableMessage"**

**Comment by Lancetnik** [comment]
@Flosckow yes, you undestood all things correctly.

About `message` VS `body`. I used `message` in `broker.publish` and `publisher.publish` interfaces due they should be used by user directly and it is explicit argument name for such usage. In the `Response` class I used `body` name due Response is whole object to publish (with body, headers and other options).

But, you point is correct and we should think about it again.
